testbed/tests/app/test_desktop.py

```diff
@@ -180,106 +182,269 @@ async def test_menu_minimize(app, app_probe):
+
+        # Enter presentation mode with a screen-window dict via the app
+        app.enter_presentation_mode(screen_window_dict)
+        # Add delay for gtk to show the windows
+        await app_probe.redraw("App is in presentation mode", delay=0.5)
+
+        assert app.is_presentation_mode
+        # All the windows should be in presentation mode.
+        for window_information in window_information_list:
+            assert (
+                window_information["window_probe"].get_window_state()
+                == WindowState.PRESENTATION
+            ), f"{window_information['window'].title}:"
+            assert (
+                window_information["window_probe"].presentation_content_size[0] > 1000
+            ), f"{window_information['window'].title}:"
+            assert (
+                window_information["window_probe"].presentation_content_size[1] > 700
+            ), f"{window_information['window'].title}:"
+
+        # Exit presentation mode
+        app.exit_presentation_mode()
+        await app_probe.redraw("App is no longer in presentation mode", delay=0.5)
+
+        assert not app.is_presentation_mode
+        for window_information in window_information_list:
+            assert (
+                window_information["window_probe"].presentation_content_size
+                == window_information["initial_content_size"]
+            ), f"{window_information['window'].title}:"
 
-        window1.show()
-        window2.show()
-        await app_probe.redraw("Extra windows are visible")
-
-        assert not app.is_full_screen
-        assert not app_probe.is_full_screen(window1)
-        assert not app_probe.is_full_screen(window2)
-        initial_content1_size = app_probe.content_size(window1)
-        initial_content2_size = app_probe.content_size(window2)
-
-        # Make window 2 full screen via the app
-        app.set_full_screen(window2)
-        await window2_probe.wait_for_window(
-            "Second extra window is full screen",
-            full_screen=True,
-        )
-        assert app.is_full_screen
+    finally:
+        for window in windows_list:
+            window.close()
+        # After closing the window, the input focus might not be on main_window.
+        # Ensure that main_window will be in focus for other tests.
+        app.current_window = main_window
+        # Add delay for gtk to show the windows
+        await app_probe.redraw("main_window is now the current window", delay=0.5)
+        assert app.current_window == main_window
 
-        assert not app_probe.is_full_screen(window1)
-        assert app_probe.content_size(window1) == initial_content1_size
 
-        assert app_probe.is_full_screen(window2)
-        assert app_probe.content_size(window2)[0] > 1000
-        assert app_probe.content_size(window2)[1] > 700
+async def test_presentation_mode_with_excess_windows_list(app, app_probe, main_window):
+    """Entering presentation mode limits windows to available displays."""
```

Comment:
Unless there's a backend-sensitive interpretation here, I don't see what this test adds. It shouldn't be possible to drive the backend API to display more windows than there are screens; as such, there's no need to test that eventuality in the testbed.

Reply:
Removed it, Thanks.